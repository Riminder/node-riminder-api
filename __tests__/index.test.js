const riminder = require('../temp/index');

const app = new riminder.Riminder({API_Key: 'api_key'});

describe('Wrapper test', () => {
  describe('Source endpoints', () => {
    test('It should call the get sources endpoint', () =>
    app.objects.getSources({}).then((response) => {
      expect(response).toMatchSnapshot();
    }));
  
    test('It should call the get source endpoint', () => {
      app.objects.getSource("id").then((response) => {
        expect(response).toMatchSnapshot();
      });
    });
  });

  describe('Profile endpoints', () => {
    test('It should call the get profiles endpoint', () => {
      const options = {
        source_ids: ["source1", "source2"],
        date_start: "1234",
        date_end: "1456",
        page: 1,
        seniority: "ALL",
        filter_id: "filter_id",
        filter_reference: "filter_reference",
        stage: "YES",
        rating: "1",
        limit: 30,
        sort_by: 'DESC',
        order_by: "RECEPTION",
      };
      app.objects.getProfiles(options)
        .then((response) => {
        expect(response).toMatchSnapshot();
        expect(response.url.query.split("&").map(entry => entry.split("=")[0])).toEqual(Object.keys(options));
      })
    });
  
    test('It should call the get profile endpoint with the profile id', () => {
      const options = {
        source_id: "source_id",
        profile_id: "profile_id",
      };
      app.objects.getProfile(options).then((response) => {
        expect(response).toMatchSnapshot();
      });
    });
  
    test('It should call the get profile endpoint with the profile reference', () => {
      const options = {
        source_id: "source_id",
        profile_reference: "profile_reference",
      };
      app.objects.getProfile(options).then((response) => {
        expect(response).toMatchSnapshot();
      });
    });
  
    test('It should call the get profile document endpoint with the profile id', () => {
      const options = {
        source_id: "source_id",
        profile_id: "profile_id",
      };
      app.objects.getProfileDocuments(options).then((response) => {
        expect(response).toMatchSnapshot();
      });
    });
  
    test('It should call the get profile document endpoint with the profile reference', () => {
      const options = {
        source_id: "source_id",
        profile_reference: "profile_reference",
      };
      app.objects.getProfileDocuments(options).then((response) => {
        expect(response).toMatchSnapshot();
      });
    });
  
    test('It should call the get profile parsing endpoint with the profile id', () => {
      const options = {
        source_id: "source_id",
        profile_id: "profile_id",
      };
      app.objects.getProfileParsing(options).then((response) => {
        expect(response).toMatchSnapshot();
      });
    });
  
    test('It should call the get profile parsing endpoint with the profile reference', () => {
      const options = {
        source_id: "source_id",
        profile_reference: "profile_reference",
      };
      app.objects.getProfileParsing(options).then((response) => {
        expect(response).toMatchSnapshot();
      });
    });
  });
});