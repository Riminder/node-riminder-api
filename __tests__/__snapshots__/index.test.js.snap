// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper test Profile endpoints It should call the get profile document endpoint with the profile id 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/documents?source_id=source_id&profile_id=profile_id",
    "path": "/sf/public/api/v1.0/profile/documents?source_id=source_id&profile_id=profile_id",
    "pathname": "/sf/public/api/v1.0/profile/documents",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_id=profile_id",
    "search": "?source_id=source_id&profile_id=profile_id",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile document endpoint with the profile reference 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/documents?source_id=source_id&profile_reference=profile_reference",
    "path": "/sf/public/api/v1.0/profile/documents?source_id=source_id&profile_reference=profile_reference",
    "pathname": "/sf/public/api/v1.0/profile/documents",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_reference=profile_reference",
    "search": "?source_id=source_id&profile_reference=profile_reference",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile endpoint with the profile id 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile?source_id=source_id&profile_id=profile_id",
    "path": "/sf/public/api/v1.0/profile?source_id=source_id&profile_id=profile_id",
    "pathname": "/sf/public/api/v1.0/profile",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_id=profile_id",
    "search": "?source_id=source_id&profile_id=profile_id",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile endpoint with the profile reference 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile?source_id=source_id&profile_reference=profile_reference",
    "path": "/sf/public/api/v1.0/profile?source_id=source_id&profile_reference=profile_reference",
    "pathname": "/sf/public/api/v1.0/profile",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_reference=profile_reference",
    "search": "?source_id=source_id&profile_reference=profile_reference",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile parsing endpoint with the profile id 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/parsing?source_id=source_id&profile_id=profile_id",
    "path": "/sf/public/api/v1.0/profile/parsing?source_id=source_id&profile_id=profile_id",
    "pathname": "/sf/public/api/v1.0/profile/parsing",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_id=profile_id",
    "search": "?source_id=source_id&profile_id=profile_id",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile parsing endpoint with the profile reference 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/parsing?source_id=source_id&profile_reference=profile_reference",
    "path": "/sf/public/api/v1.0/profile/parsing?source_id=source_id&profile_reference=profile_reference",
    "pathname": "/sf/public/api/v1.0/profile/parsing",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_reference=profile_reference",
    "search": "?source_id=source_id&profile_reference=profile_reference",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profiles endpoint 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profiles?source_ids=[source1,source2]&date_start=1234&date_end=1456&page=1&seniority=ALL&filter_id=filter_id&filter_reference=filter_reference&stage=YES&rating=1&limit=30&sort_by=DESC&order_by=RECEPTION",
    "path": "/sf/public/api/v1.0/profiles?source_ids=[source1,source2]&date_start=1234&date_end=1456&page=1&seniority=ALL&filter_id=filter_id&filter_reference=filter_reference&stage=YES&rating=1&limit=30&sort_by=DESC&order_by=RECEPTION",
    "pathname": "/sf/public/api/v1.0/profiles",
    "port": null,
    "protocol": "https:",
    "query": "source_ids=[source1,source2]&date_start=1234&date_end=1456&page=1&seniority=ALL&filter_id=filter_id&filter_reference=filter_reference&stage=YES&rating=1&limit=30&sort_by=DESC&order_by=RECEPTION",
    "search": "?source_ids=[source1,source2]&date_start=1234&date_end=1456&page=1&seniority=ALL&filter_id=filter_id&filter_reference=filter_reference&stage=YES&rating=1&limit=30&sort_by=DESC&order_by=RECEPTION",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Source endpoints It should call the get source endpoint 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/source?source_id=id",
    "path": "/sf/public/api/v1.0/source?source_id=id",
    "pathname": "/sf/public/api/v1.0/source",
    "port": null,
    "protocol": "https:",
    "query": "source_id=id",
    "search": "?source_id=id",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Source endpoints It should call the get sources endpoint 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/sources",
    "path": "/sf/public/api/v1.0/sources",
    "pathname": "/sf/public/api/v1.0/sources",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;
